src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -143,6 +143,9 @@ module.exports = function<T, P, I, TI, C>(config : HostConfig<T, P, I, TI, C>) :
       root.pendingContext = getContextForSubtree(parentComponent);
       // TODO: Use pending work/state instead of props.
       root.current.pendingProps = element;
+      if (root.current.alternate) {
```

Comment:
I'll add a comment to explain. This is so that updates that occur after a root completes but before it commits are not dropped. I added a unit test for this. Previously completed roots were always immediately committed so it wasn't needed. (Btw this is the thing I joked about on Twitter that took me two hours to debug haha.)